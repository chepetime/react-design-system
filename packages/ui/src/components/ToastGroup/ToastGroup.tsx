import React from "react";
import classnames from "classnames";
import css from "./ToastGroup.module.scss";

export interface ToastGroupProps extends React.ComponentPropsWithRef<"div"> {
  children?: React.ReactNode;
}

export const ToastGroup = React.forwardRef<HTMLDivElement, ToastGroupProps>(
  (props: ToastGroupProps, ref: React.Ref<HTMLDivElement>) => {
    return (
      <div className={classnames(css.ToastGroup)} ref={ref}>
        <p>ToastGroup component working!</p>
        <div>{props.children}</div>
      </div>
    );
  }
)

ToastGroup.displayName = "ToastGroup";
