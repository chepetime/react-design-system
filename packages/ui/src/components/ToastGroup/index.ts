export { ToastGroup } from "./ToastGroup";
export type { ToastGroupProps } from "./ToastGroup";