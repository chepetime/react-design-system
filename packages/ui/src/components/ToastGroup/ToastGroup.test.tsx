
import React from "react";
import userEvent from "@testing-library/user-event";
import "@testing-library/jest-dom";
import { ToastGroup } from "./ToastGroup";
import { render, screen } from "@testing-library/react";

describe("ToastGroup", () => {
  it("renders without errors", () => {
    render(<ToastGroup />);
    // Assert that the component renders without throwing any errors
  });

  it("renders with children", () => {
    render(<ToastGroup>Some children</ToastGroup>);
    // Assert that the component renders the provided children
    const childrenElement = screen.getByText("Some children");
    expect(childrenElement).toBeInTheDocument();
  });

  it("handles user interactions", () => {
    render(<ToastGroup>Some children</ToastGroup>);
    // Simulate user interactions with the component
    userEvent.click(screen.getByText("Some children"));
    // Assert the expected outcome of the user interaction
    // For example, check if a certain state has changed or an event handler has been called
  });

  it("matches snapshot", () => {
    const { container } = render(<ToastGroup />);
    // Assert that the rendered component matches a previously captured snapshot
    expect(container).toMatchSnapshot();
  });
});
