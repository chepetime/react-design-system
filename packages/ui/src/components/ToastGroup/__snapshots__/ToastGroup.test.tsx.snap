// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ToastGroup matches snapshot 1`] = `
<div>
  <div
    class="ToastGroup"
  >
    <p>
      ToastGroup component working!
    </p>
    <div />
  </div>
</div>
`;
